table/tables/cache_test.go

```diff
@@ -111,5 +113,51 @@ func TestCacheTableBasicScan(t *testing.T) {
 		tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1105 IndexMerge is inapplicable or disabled"))
 	}
 	assertSelect()
+}
+
+func TestCacheTableBasicReadAndWrite(t *testing.T) {
+	t.Parallel()
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk1 := testkit.NewTestKit(t, store)
+	tk.MustExec("drop table if exists write_tmp1")
+	tk.MustExec("create  table write_tmp1 (id int primary key auto_increment, u int unique, v int)")
+	tk.MustExec("insert into write_tmp1 values" +
+		"(1, 101, 1001), (3, 113, 1003)",
+	)
+	tk.MustExec("alter table write_tmp1 cache")
+	// Read and add read lock
+	tk.MustQuery("select *from write_tmp1").Check(testkit.Rows("1 101 1001",
+		"3 113 1003"))
 
+	time.Sleep(1)
+	// read lock should valid
+	result := tk.MustQuery("explain format = 'brief' select *from write_tmp1")
+	result.Check(testkit.Rows("UnionScan 10000.00 root  ",
+		"└─TableReader 10000.00 root  data:TableFullScan",
+		"  └─TableFullScan 10000.00 cop[tikv] table:write_tmp1 keep order:false, stats:pseudo"))
+	fmt.Println(result)
+	tk1.MustExec("use test")
+	tk1.MustExec("insert into write_tmp1 values (2, 222, 222)")
+	// none lock can't read cache first wait add read lock
+	tk.MustQuery("explain format = 'brief' select *from write_tmp1").Check(testkit.Rows(
+		"TableReader 10000.00 root  data:TableFullScan",
+		"└─TableFullScan 10000.00 cop[tikv] table:write_tmp1 keep order:false, stats:pseudo"))
+	time.Sleep(3 * time.Second)
```

Comment:
Slow down the CI a lot.... what's the purpose of this sleep?

Reply:
i think we should... wait the read lease expire...